src/bika/lims/api/__init__.py

```diff
@@ -718,15 +718,19 @@ def get_icon(brain_or_object, html_tag=True):
     # work for Contents coming from other catalogs than the
     # `portal_catalog`
     portal_types = get_tool("portal_types")
-    fti = portal_types.getTypeInfo(brain_or_object.portal_type)
+    portal_type = thing if is_string(thing) else get_portal_type(thing)
```

Comment:
I would rather use `is_object` instead of just checking for string type:
https://github.com/senaite/senaite.core/blob/2.x/src/bika/lims/api/__init__.py#L400

```python
if api.is_object(thing):
    portal_type = api.get_portal_type(thing)

Reply:
Addressed with https://github.com/senaite/senaite.core/pull/2458/commits/56bf2b3f10a14a6048984c881edf48d2e9c2991b